File: app/src/main/java/ch/bailu/aat/gpx/xml_parser/parser/gpx/EleParser.java
package ch.bailu.aat.gpx.xml_parser.parser.gpx;

import org.xmlpull.v1.XmlPullParser;
import org.xmlpull.v1.XmlPullParserException;

import java.io.IOException;

import ch.bailu.aat.gpx.xml_parser.parser.TagParser;
import ch.bailu.aat.gpx.xml_parser.scanner.Scanner;

public class EleParser extends TagParser {
    public EleParser() {
        super("ele");
    }

    @Override
    public void parseText(XmlPullParser parser, Scanner scanner) throws IOException, XmlPullParserException {
        scanner.altitude.scan(parser.getText());
    }

    @Override
    protected void parseAttributes(XmlPullParser parser, Scanner scanner) {

    }

    @Override
    public boolean parseTags(XmlPullParser parser, Scanner scanner) throws IOException, XmlPullParserException {
        return false;
    }

    @Override
    public void parsed(XmlPullParser parser, Scanner scanner) {

    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/xml/parser/gpx/EleParser.kt
package ch.bailu.aat_lib.xml.parser.gpx

import ch.bailu.aat_lib.xml.parser.osm.TagParser
import ch.bailu.aat_lib.xml.parser.scanner.Scanner
import org.xmlpull.v1.XmlPullParser
import java.io.IOException

class EleParser : TagParser("ele") {
    @Throws(IOException::class)
    public override fun parseText(parser: XmlPullParser, scanner: Scanner) {
        scanner.altitude.scan(parser.text)
    }

    override fun parseAttributes(parser: XmlPullParser, scanner: Scanner) {}
    public override fun parseTags(parser: XmlPullParser, scanner: Scanner): Boolean {
        return false
    }

    public override fun parsed(parser: XmlPullParser, scanner: Scanner) {}
}
